src/crush/crush.h

```diff
@@ -524,6 +525,14 @@ static inline const char *crush_alg_name(int alg)
 	}
 }
 
+// Crush errors
+struct crush_errors_t {
+  uint64_t choose_firstn_tries_exceeded_errors;
+  uint64_t choose_indep_tries_exceeded_errors;
+  bool has_errors;
+  bool new_errors;
```

Comment:
I'm not following why we need these two bools.  It seems like something like this would only be needed/useful if you were making multiple calls with the same crush_errors_t with has_errors=true and new_errors=false, but I'm not sure why we would need to do that.  Why not just use an empty crush_errors_t in the cases where we need to see if there are new instead of old errors?

We could add a uint64_t total() { return ... + ...; } method and use that to see whether there were any errors, use a separate one of these to sum up errors on a per-pool basis in the mapping phase, and so on.